runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -282,6 +283,14 @@ public DirectPipelineResult run(Pipeline pipeline) {
     Collection<ModelEnforcementFactory> parDoEnforcements = createParDoEnforcements(options);
     enforcements.put(ParDo.Bound.class, parDoEnforcements);
     enforcements.put(ParDo.BoundMulti.class, parDoEnforcements);
+    if (options.isTestEncodability()) {
+      enforcements.put(
```

Comment:
I'm curious why only these 3 transforms need encodability enforcement. Is it because they are the only primitives which can mutate elements? Might be nice to have a test which shows enforcement being applied to a non-primitive, i.e. MapElements.